base/server/python/pki/server/cli/cert.py

```diff
@@ -990,3 +990,230 @@ def execute(self, argv):
+                # subsystem name to the ID
+                for cert in certs:
+                    if cert['id'] != 'sslserver' and cert['id'] != 'subsystem':
+                        cert['id'] = subsystem.name + '_' + cert['id']
+
+                    # Append only unique certificates to other subsystem certificate list
+                    # ca_signing isn't supported yet
+                    if cert['id'] in fix_certs or cert['id'] == 'ca_signing':
+                        continue
+
+                    fix_certs.append(cert['id'])
+
+        logger.info('Fixing the following certs: %s', fix_certs)
+
+        # 2. Stop the server if it's up
+        instance.stop()
+
+        # 3. Find the subsystem and disable Self-tests
+        try:
+            # Placeholder used to hold subsystems whose selftest have been turned off
+            target_subsys = []
+
+            if any(cert_id in fix_certs for cert_id in ('sslserver', 'subsystem')):
+                # If the cert is either sslserver/subsystem, disable selftest for all
+                # subsystems since all subsystems use these 2 certs.
+                for subsystem in instance.subsystems:
+                    target_subsys.append(subsystem)
+
+            else:
+                for cert_id in fix_certs:
+                    # Since we already filtered sslserver/subsystem we can be quite sure
+                    # that this split will definitely be of form: <subsys>_<cert_tag>
+                    subsystem_name = cert_id.split('_', 1)[0]
+                    subsystem = instance.get_subsystem(subsystem_name)
+
+                    # If the subsystem is wrong, stop the process
+                    if not subsystem:
+                        logger.error('No %s subsystem in instance %s.',
+                                     subsystem_name, instance_name)
+                        sys.exit(1)
+
+                    target_subsys.append(subsystem)
+
+            # Remove duplicate subsystem entries, if any.
+            # Example of duplicates:
+            # fix_certs = [ca_ocsp_signing, ca_audit_signing] -> will add 'ca' entry twice
+            target_subsys = list(set(target_subsys))
+
+            for subsystem in target_subsys:
+                subsystem.set_startup_test_criticality(False)
+                subsystem.save()
+
+            logger.info('Selftests disabled for subsystems: %s', ', '.join(
+                str(x.name) for x in target_subsys))
+
+            # 4. Bring up the server using a temp SSL cert
+            # 4a. Create temp SSL cert
+            instance.cert_create(cert_id='sslserver', temp_cert=True)
+
+            # 4b. Delete the existing SSL Cert
+            target_subsys[0].cert_del('sslserver')
+
+            # 4d. Import it into NSS DB (since sslserver is used by ALL subsystems, you can
+            # just use the first subsystem in target_subsys)
+            cert = target_subsys[0].nssdb_import_cert('sslserver')
```

Comment:
This would cause confusion as we discussed earlier: https://github.com/dogtagpki/pki/pull/67#discussion_r225696926

Reply:
The PKIInstance.cert_import() probably can import into nssdb then call cert_update_config() to update the CS.cfg. We can always change the method name if it's confusing, but I think the functionality to import a cert into nssdb belongs to PKIInstance (since it owns the nssdb), and the PKISubsystem only owns the CS.cfg.